File: ExtremeParser/Parsers/PawnDoc.vb
Imports System.Text.RegularExpressions

Public Class PawnDoc
    Public Shared Sub Parse(ByRef code As String, fileName As String, ByRef parts As CodeParts, Optional add As Boolean = True)
        For Each Match As Match In Regex.Matches(code, "/\*\*([\s\S]*?)\*/")
            Dim val As String = Match.Groups(1).Value 'Group 0 which contains only the internal text without the start and the end.

            If Not val.Contains("<summary>") Or Not val.Contains("</summary>") Then Continue For

            Try
                Dim pwn = New PawnDocParser(val)
                If add Then
                    parts.pawnDocs.Add(pwn)
                Else
                    parts.pawnDocs.RemoveAll(Function(x) x.Summary = pwn.Summary)
                End If
            Catch ex As ParserException
                Throw New ParserException(ex.Message, "")
            End Try
        Next
    End Sub
End Class

File: ExtremeParser/Parsers/PawnDoc.cs
using System;
using System.Text.RegularExpressions;
using ExtremeParser.Exceptions;

namespace ExtremeParser.Parsers
{
    public class PawnDoc
    {
        public static void Parse(string code, string fileName, CodeParts parts, bool add = true)
        {
            foreach (Match match in Regex.Matches(code, "\\/\\*\\*([^*]*(?:\\*(?!\\/)[^*]*)*)\\*\\/"))
            {
                string val = Convert.ToString(match.Groups[1].Value); //Group 0 which contains only the internal text without the start and the end.
			
                if (!val.Contains("<summary>") || !val.Contains("</summary>"))
                {
                    continue;
                }
			
                try
                {
                    var pwn = new PawnDocParser(val);
                    if (add)
                    {
                        parts.PawnDocs.Add(pwn);
                    }
                    else
                    {
                        parts.PawnDocs.RemoveAll(x => x.Summary == pwn.Summary);
                    }
                }
                catch (ParserException ex)
                {
                    throw (new ParserException(ex.Message, ""));
                }
            }
        }
    }
}
